app/src/main/java/org/wikipedia/page/PageFragment.java

```diff
@@ -917,17 +924,45 @@ private void setupMessageHandlers() {
         };
         bridge.addListener("link_clicked", linkHandler);
 
-        bridge.addListener("reference_clicked", new ReferenceHandler() {
-            @Override
-            protected void onReferenceClicked(int selectedIndex, @NonNull List<Reference> adjacentReferences) {
+        bridge.addListener("reference_clicked", (String messageType, JSONObject messagePayload) -> {
+            if (!isAdded()) {
+                L.d("Detached from activity, so stopping reference click.");
+                return;
+            }
 
-                if (!isAdded()) {
-                    L.d("Detached from activity, so stopping reference click.");
-                    return;
-                }
+            disposables.add(ServiceFactory.getRest(getTitle().getWikiSite()).getReferences(getTitle().getConvertedText())
+                    .subscribeOn(Schedulers.io())
```

Comment:
Could we actually store/cache the References response in the fragment, so that we don't have to make the network call every time?

Reply:
Sure.